server/src/main/java/org/opensearch/snapshots/SnapshotUtils.java

```diff
@@ -89,8 +103,8 @@ public static List<String> filterIndices(List<String> availableIndices, String[]
                         result = new HashSet<>();
                     }
                 } else if (indexOrPattern.charAt(0) == '-') {
-                    // if its the first, fill it with all the indices...
-                    if (i == 0) {
+                    // if its the first index pattern, fill it with all the indices...
+                    if (i == 0 && selectedIndices.length == 1) {
```

Comment:
What if selectedIndices has multiple, only negative patterns?

i.e. `selectedIndices = ['-.opendistro_security', '-.kibana_1']`

Shouldn't this be interpreted as all indices except for these 2? When adding to the condition here to say only add all available indices if the size is 1 then it will not support this scenario. A workaround would be to have selectedIndices be  `['*', '-.opendistro_security', '-.kibana_1']`

Reply:
Oops I did not mean to leave the length == 1 catch in. Good catch.

Reply:
I removed it since now it will only ever do the negated index --> " wildcard minus this one " if all the indices are negated.